trafficmesh/signature.go

```diff
@@ -0,0 +1,65 @@
+package trafficmesh
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+const signatureHeader = "X-NF-Mesh-Signature"
+
+type SignatureDecoder struct {
+	secret string
+}
+
+type SignaturePayload struct {
+	jwt.StandardClaims
+	SiteID    string `json:"sid,omitempty"`
+	DeployID  string `json:"did,omitempty"`
+	AccountID string `json:"aid,omitempty"`
+	URL       string `json:"url,omitempty"`
+	Remapped  bool   `json:"remapped,omitempty"`
+}
+
+func NewSignatureDecoder(secret string) *SignatureDecoder {
```

Comment:
this could use a comment that passing an empty string for secret is valid and will make `DecodeSignature` a no-op